mobile/src/main/java/org/openhab/habdroid/ui/activity/ContentController.java

```diff
@@ -543,99 +554,115 @@ protected static int determineTransition(FragmentUpdateReason reason) {
     public static class CommunicationFailureFragment extends StatusFragment {
```

Comment:
The whole reason for the existence of those classes is/was the different onClick implementation. Could nuke them with the current impl. What I think I'd prefer, though, is
- having action implementation methods in the base class (`enableWifi()`, `enableDemoMode()` etc.)
- keep the onClick implementations in the sub classes and make those call the base class methods
- nuke the button tags

That way it's more obvious what's going on (less indirections), and the argument list for the fragment becomes shorter.

Reply:
I added the tags, because in MissingConfigurationFragment the buttons can have different actions based on hasWifiEnabled and resolvedAttempted. I can re-add the onClick() functions in all fragments, but then I need to split MissingConfigurationFragment in three fragments.

Reply:
You don't ;-) I'll upload a suggestion to my repo.

Reply:
https://github.com/maniac103/openhab.android/commit/dee4267e5022a335df14f3f442649fe41da5ce48 would be my suggestion.